app/src/main/java/app/revanced/manager/ui/screen/PatcherScreen.kt

```diff
@@ -85,6 +88,16 @@ fun PatcherScreen(
         }
     }
 
+    if (patcherSucceeded == null) {
+        DisposableEffect(Unit) {
+            val window = (context as Activity).window
+            window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
```

Comment:
Most examples I have seen for applying flags seem to do it with a lifecycle observer

Reply:
Really?
I searched for how to add `FLAG_KEEP_SCREEN_ON` flag in Jetpack Compose and this came up.

Additionally, the flag needs to be cleared when the patching succeeds or fails, so it should have nothing to do with lifecycle.

Reply:
Hmm, some examples I've seen for `FLAG_SECURE` seem to use lifecycle observer.  Like the second comment [here](https://stackoverflow.com/questions/78694809/is-there-a-way-to-apply-layoutparams-flag-secure-to-compose-screen-level-to-prev). Does the flag stick around if you exit the app and then go back in?

Reply:
> Does the flag stick around if you exit the app and then go back in?

Yes.
When I leave the app while patching, screen timeout will be activated, and when I back to the app, screen timeout will be disabled again.

This is also described here:
> If an app with the `FLAG_KEEP_SCREEN_ON` flag goes into the background, the system allows the screen to turn off normally. You don't need to explicitly clear the flag in this case.
https://developer.android.com/develop/background-work/background-tasks/awake/screen-on